pkg/process/process_manager.go

```diff
@@ -104,6 +136,61 @@ func (pm *Manager) startMonitoring() {
 	}
 }
 
+func (pm *Manager) checkMountPointStatusForEngine() {
+	var processToUpdate []*Process
+	volumeMountPointMap, err := getVolumeMountPointMap()
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to get all volume mount points")
+	}
+
+	pm.lock.RLock()
+	for _, p := range pm.processes {
+		p.lock.Lock()
+		if isEngineProcess(p) && p.State == StateRunning {
+
+			nameSlices := strings.Split(p.Name, "-")
+			volumeNameSHA := sha256.Sum256([]byte(strings.Join(nameSlices[:len(nameSlices)-2], "-")))
+
+			if mp, exists := volumeMountPointMap[hex.EncodeToString(volumeNameSHA[:])]; exists {
+				p.Conditions[types.EngineConditionFilesystemReadOnly] = util.IsMountPointReadOnly(mp)
+				processToUpdate = append(processToUpdate, p)
+			}
+		}
+		p.lock.Unlock()
+	}
+	pm.lock.RUnlock()
+
+	for _, p := range processToUpdate {
+		p.UpdateCh <- p
+	}
+
+}
+
+func isEngineProcess(p *Process) bool {
+	return p.PortCount == DefaultEnginePortCount
```

Comment:
This is a weak check. It works now, but I suggest differentiating process type. Let's have a ticket for it later.

Reply:
We also use it in longhorn-maanger: https://github.com/longhorn/longhorn-manager/blob/master/engineapi/instance_manager.go#L290
but I agree we should have more reliable check

Reply:
https://github.com/longhorn/longhorn/issues/7393